exporter/signalfxexporter/exporter.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfxexporter
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumererror"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf"
+	"go.uber.org/zap"
+)
+
+// New returns a new SignalFx exporter.
+func New(
+	config *Config,
+	logger *zap.Logger,
+) (exporter.MetricsExporter, error) {
+
+	if config == nil {
+		return nil, errors.New("nil config")
+	}
+
+	if config.Name() == "" {
+		config.SetType(typeStr)
+		config.SetName(typeStr)
+	}
```

Comment:
Even if we don't expose `New` but only the factory methods it will still be possible to use `SetType` and `SetName` to put empty values which is not good in terms of logging and metrics - the way to protect from this will be to change the interfaces (perhaps worth considering). That said this is being a bit paranoid about such usage since it doesn't make much sense to set these to empty nor should it happen when used in the collector.